compiler/test/dotty/tools/scripting/BashExitCodeTests.scala

```diff
@@ -0,0 +1,158 @@
+
+  /*
+   * Compiles the class checking exit code
+   *
+   * @param testName name of the test containing the extension
+   * @param expectedExitCode expected exit code from the output
+   * @param deleteTempDir if temporary directory created for compilation output should delete in this scope
+   *
+   * @return Created temporary directory
+   */
+  private def testCompilationExitCode(testName: String, expectedExitCode: Int, deleteTempDir: Boolean = false): File =
+    val temporaryDir = Files.createTempDirectory(testName)
+    assertTestExists(testName) { testFile =>
+      try {
+        val testFilePath = testFile.absPath
+        val commandline = (Seq(scalacPath, "-d", temporaryDir, testFilePath)).mkString(" ")
+        val (validTest, exitCode, _, _) = bashCommand(commandline)
+        if verifyValid(validTest) then
+          assertEquals(expectedExitCode, exitCode)
+      } finally {
+        if deleteTempDir then Util.deleteFile(temporaryDir.toFile)
+      }
+    }
+    temporaryDir.toFile
+
+  /*
+   * Runs the command and checks the exit code
+   *
+   * @param args arguments for command line
+   * @param expectedExitCode expected exit code from the output
+   */
+  private def testCommandExitCode(args: Seq[String], expectedExitCode: Int): Unit =
+    val commandline = args.mkString(" ")
+    val (validTest, exitCode, output, erroutput) = bashCommand(commandline)
+    if verifyValid(validTest) then
+      assertEquals(expectedExitCode, exitCode)
+
+
+  /* Compiles and then runs the code checking the exit code
+   *
+   * @param testFileName name of the test
+   * @param runExitCode expected exit code from the runner output
+   */
+  private def testCompileAndRun(testFileName: String, runExitCode: Int): Unit =
+    val outputDirectory = testCompilationExitCode(testFileName, 0)
+
+    def testRuntimeExitCode(className: String, expectedExitCode: Int): Unit =
+        val testClassFile = outputDirectory.files.find(_.getName == s"$className.class")
+        assert(testClassFile.isDefined)
+        val commandline = (Seq(scalaPath, "-classpath", outputDirectory.getAbsolutePath, className)).mkString(" ")
+        val (validTest, exitCode, o, e) = bashCommand(commandline)
+          if verifyValid(validTest) then
+            assertEquals(expectedExitCode, exitCode)
+
+    try {
+      val generatedClassName = testFileName.split('.').head
+      testRuntimeExitCode(generatedClassName,  runExitCode)
+    } finally {
+      Util.deleteFile(outputDirectory)
+    }
+
+  /*
+   * Checks if scripting test resources contains test with given `testName`
+   * And then runs function `test`
+   *
+   * @param testName name of the test containing the extension
+   * @param test check to be run on found test file
+   */
+  private def assertTestExists(testName: String)(test: File => Unit) =
+    val file = testFiles.find(_.getName == testName)
+    assert(file.isDefined)
+    test(file.get)
+
+class BashExitCodeTests:
```

Comment:
```suggestion
@Category(Array(classOf[BootstrappedOnlyTests]))
class BashExitCodeTests:
```

These tests depend on the output of `dist/pack` which uses the bootstrapped compiler, but we don't want to force bootstrapping in `test_non_bootstrapped` (see #15513).

You'll need to add an import as well: `import org.junit.experimental.categories.Category`

Reply:
Yeah, thanks for noticing